### ISSUE TITLE
resuming after tts announcement 

### ISSUE THREAD

**Issue by hmorais29** [music-assistant/hass-music-assistant#697]
### What version of Music Assistant has the issue?

2022.7.0b7

### The problem

i noticed that if i send a tts to a speaker (google nest) playing radio from tune in which i previuous ask it to play (by voice command) the tts is played in that speaker and then it resumes what i was previously listening via music assistance and not what it was playin when it was interrupted by the tts annoucement.

My guess is that when i ask to play radio by voice, the device used is not the one created by the music assistant, althoug if i look at music assistant player i can see the music being played in the device created by music assistant

### How to reproduce

Ask a speaker to play radio by a voice command (hey google play radio xxx) then send a tts notification to that speaker.
The annoucement will interrupt the radio and when it resumes it starts playing spotify (that i was listening before the radio station)

### Relevant log output

N/A

### What version of Home Assistant Core are your running

2022.7.5

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by erkr** [comment]
Hi, I think you are right that the radio is not playing from MA.
You can easily check that. When content is playing from other applications, the player in MA shows a square stop symbol, while it shows play or pause buttons when MA is in control.

**Comment by marcelveldt** [comment]
I'm not sure if we can even resume content that is not instantiated by MA but I can give it a try

**Comment by erkr** [comment]
I think that is not needed for now. Maybe just a switch to allow TTS on mass players only when MA is in control or idle, makes sense.

**Comment by marcelveldt** [comment]
It should be possible if the "media_content_id" attribute of the player contains an url/uri that makes sense.
So please investigate what the value of "media_content_id" is while playing from another source. 
Next step is what happens if you issue a "play_media" service call with that exact uri. If that works, MA could try to send that uri as some sort of best-effort approach to resume

**Comment by hmorais29** [comment]
when i ask via voice the speaker to play radio the media_conten_id is 
"media_content_id: http://mcrscast1.mcr.iol.pt/comercial.mp3"

and when i called the service play_media 

```
service: media_player.play_media
data:
  media_content_id: http://mcrscast1.mcr.iol.pt/comercial.mp3
  media_content_type: music
  announce: true
  entity_id: media_player.mass_despertador_lili
```

it works for ~10seconds  and i found this in the logs

```
Logger: music_assistant.players
Source: /usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py:323
First occurred: 09:24:04 (3 occurrences)
Last logged: 09:25:52

Timeout while playing alert
```

not sure if it helps somehow

**Comment by marcelveldt** [comment]
do not set announce to true because that is the announce feature ;-)

good news is that it is actually possible to resume. I'll add this in the code.

**Comment by hmorais29** [comment]
> do not set announce to true because that is the announce feature ;-)

you are right i forgot to remove the announce, silly me 🤦

**Comment by marcelveldt** [comment]
Fix available in the 2022.7 release!
Please reopen (or create a new issue) if you still may experience issues. Thanks!

**Comment by hmorais29** [comment]
hi again @marcelveldt 

I tested and the resume is working although it takes some time (not an issue) but i receive an error in the logs everytime the tts is sent tried 3 times just to be sure

`Logger: homeassistant
Source: components/cast/media_player.py:737
First occurred: 12:51:55 (3 occurrences)
Last logged: 12:57:42

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 543, in play_index
    await self.player.play_url(stream.url)
  File "/config/custom_components/mass/player_controls.py", line 252, in play_url
    await self.entity.async_play_media(
  File "/usr/src/homeassistant/homeassistant/components/cast/media_player.py", line 737, in async_play_media
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/controllers/media.py", line 547, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError`


like you said, should i raise a new issue? do you think it worth the try ?

**Comment by marcelveldt** [comment]
Did you test with version 2022.7.1 ?

**Comment by hmorais29** [comment]
yep, that is the one i'm using now

**Comment by marcelveldt** [comment]
That error originates from the pychromecast itself because TuneIn streams are very slow to resume.
I strongly suggest you to use Music Assistant to play your radio streams because all that is handled much better than in default HA integration

**Comment by marcelveldt** [comment]
Can you confirm that the resume is fast (and without the error) when the radio/music was started from MA ?

**Comment by hmorais29** [comment]
yes, it seems to take around the same time to resume but it shows a different error

`Logger: music_assistant.players
Source: /usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py:368
First occurred: 13:19:09 (1 occurrences)
Last logged: 13:19:09

Error while playing announcement
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 365, in play_announcement
    await self._wait_for_state(PlayerState.PLAYING, silence_item.item_id)
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 917, in _wait_for_state
    raise TimeoutError(f"Timeout while waiting on state(s) {state}")
TimeoutError: Timeout while waiting on state(s) (<PlayerState.PLAYING: 'playing'>,)`

**Comment by marcelveldt** [comment]
is this a cast group or regular player ?

**Comment by hmorais29** [comment]
it's regular one

**Comment by marcelveldt** [comment]
and you are 100% sure you updated to 2022.7.1. and restarted HA after that to apply the update?

**Comment by hmorais29** [comment]
well i am, but i'll give it another go and retest just in case

**Comment by marcelveldt** [comment]
thanks

**Comment by marcelveldt** [comment]
I can reproduce the issue. I'll try to fix it

**Comment by hmorais29** [comment]
thanks, i just confirmed as well after the restart that it still happens

**Comment by marcelveldt** [comment]
@hmorais29 
Fixed the issue in 2022.7.2 Please test with that patch release, thanks!


@lucasimons your issue seems to be different than the one reported here and seems to come from the linkplay component itself.

**Comment by hmorais29** [comment]
> @hmorais29 Fixed the issue in 2022.7.2 Please test with that patch release, thanks!
> 
> @lucasimons your issue seems to be different than the one reported here and seems to come from the linkplay component itself.

sure, i'll give it a test once released 😃

**Comment by hmorais29** [comment]
@marcelveldt 
just tested it, it seems perfect, way faster and without errors.
tested a couple of times either when palying radio directly though MA either by voice command directly to the speaker.. worked like a charm
